com.unity.ml-agents/Runtime/Actuators/IActuator.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using Unity.MLAgents.Policies;
+
+namespace Unity.MLAgents.Actuators
+{
+    /// <summary>
+    /// Abstraction that facilitates the execution of actions.
+    /// </summary>
+    public interface IActuator
+    {
+        float[] Actions
+        {
+            get;
+        }
+
+        void UpdateActions(float[] fullActionBuffer, int offset);
+
+        /// <summary>
+        /// Get the number of actions this IActuator will execute on.
+        /// </summary>
+        int GetNumberOfActions();
```

Comment:
I think we also need to know the space type of the actions. Maybe we could combine this and GetBranchSizes, and return a struct like
```
struct ActuatorSpace {
  SpaceType[] m_SpaceTypes;
  int[] m_BranchSizes

  public static ActuatorSpace MakeContinuous(int numActions)
  {
    var actSpace = new ActuatorSpace {m_SpaceTypes = new SpaceType[numActions];
    // intialize all actSpace.m_SpaceTypes to SpaceType.Continuous
    return actSpace;
  }
  public static ActuatorSpace MakeDiscrete(int[] m_BranchSize)
  {
  }
  // private constructor?
}
```
Right now, we need to be either all continuous or all discrete, but hopefully we can relax that in the future (on the trainer side), and this struct would let us keep the interface consistent then.

Reply:
Sounds good.